### ISSUE TITLE
t.news.nntp.NNTPServer contains dead code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5537]
|<img alt="rockstar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| rockstar reported|
|-|-|
|Trac ID|trac#5537|
|Type|enhancement|
|Created|2012-03-13 05:08:15Z|
|Branch|https://github.com/twisted/twisted/tree/nntpserver-dead-code-5537|

do_XROVER is declared twice, resulting in dead code.  The first declaration should be removed along with its dependent private methods.

Attachments:

* [5537.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/274/2746072afbcd8668ddca24f8898168ab16b6142d/da9aa3fdbb126e2ccc81a58f882e7d68fd35d963.patch) (1541 bytes) - added by rockstar on 2012-03-13 05:36:44Z - Patch to remove dead code
* [5537.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/274/2746072afbcd8668ddca24f8898168ab16b6142d/00cb4f3a1b51e18dc36a0bf5c42c5038a7a86645.patch) (4235 bytes) - added by rockstar on 2012-03-14 07:21:32Z - New patch with jesstess's requested changes

<details><summary>Searchable metadata</summary>

```
trac-id__5537 5537
type__enhancement enhancement
reporter__rockstar rockstar
priority__normal normal
milestone__ 
branch__branches_nntpserver_dead_code_5537 branches/nntpserver-dead-code-5537
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__news news
keywords__ 
time__1331615295000000 1331615295000000
changetime__1332183887000000 1332183887000000
version__None None
owner__exarkun exarkun
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="rockstar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|rockstar set owner to rockstar<br>rockstar set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="rockstar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|rockstar removed owner<br>rockstar set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to rockstar|
|-|-|

Thanks for working on this, rockstar! A couple of comments:

 * The two versions of `do_XROVER` had different function signatures. http://www.mibsoftware.com/userkt/nntpext/0028.htm suggests that you only need a range, so the one you leave seems reasonable, but this does make it clear that we need some unit tests for `do_XROVER`. Can you add some as part of this ticket?
 * Can you add a NEWS file as described in http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles

Thanks again!

**Comment by twisted-trac** [comment]
|<img alt="rockstar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|rockstar commented|
|-|-|

It seemed prudent to add more tests around the NNTPServer since I was working on that stuff anyway.  If that's a problem, I'd be happy to re-submit the patch and open another ticket for the extra tests.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [33813]) Branching to 'nntpserver-dead-code-5537'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [33814]) Apply 5537.2.patch 

refs [#5537](https://github.com/twisted/twisted/issues/5537)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

Thanks!  A few minor points:

  1. I _think_ http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles is pretty clear about how "news files" or "news fragments" work.  However, the name "news file" (or "NEWS file") can be quite misleading, so apologies for that.  What jesstess was actually asking for in her review was a file like `twisted/news/topfiles/5537.misc`.  These and others get aggregated into the actual "NEWS" file at release time.
  1. The new unit tests are great.  We always want more of those. :)  A few things about them:
    1. Test methods should have docstrings explaining the feature being verified.
    1. Some extra `transport.clear()` calls would let you avoid repeating the "200 server ready - posting allowed" message in most of the tests.

I think I remember some stuff about NNTP, perhaps I'll give these points a go.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

(In [33816]) Back out NEWS file change

refs [#5537](https://github.com/twisted/twisted/issues/5537)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Did some of those things.  Rushing this back to review since I have to close my laptop, hopefully it's cool though.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_13"></a>|@dreid set owner to @exarkun|
|-|-|

1.do_XROVER should lose the spaces around the '=' in it's keyword arguments.
  1. It'd also be nice if do_XROVER didn't call it's argument range (shadowing the builtin range).  It looks like do_%s methods are never called with keyword arguments only positional arguments so the keyword argument is just there to provide a default value.
  1. do_XROVER should probably also get a docstring?

 1. test_list/group/listgroup/xrover should be test_LIST/GROUP/LISTGROUP/XROVER this makes it obvious in the test output what they're doing and makes them not look like PEP8 names.
 1. twisted/news/topfiles/5537.news is empty.

Thanks for working on this guys.  News has had no significant changes in 3 releases.  I think at this point any change counts as significant so good job.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun removed owner|
|-|-|

Thanks, addressed those points, except for renaming the `range` parameter.  The name is used at least one other place in this module.  I think it'd be good to continue to expand the test coverage for this code, then fix the name in both places.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_15"></a>|@dreid set owner to @exarkun|
|-|-|

Replying to [exarkun](#note_14):
> Thanks, addressed those points, except for renaming the `range` parameter.  The name is used at least one other place in this module.  I think it'd be good to continue to expand the test coverage for this code, then fix the name in both places.

Ok, that sounds like a reasonable strategy.  You should go ahead and merge this after it passes a build.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun set status to `closed`|
|-|-|

(In [33941]) Merge nntpserver-dead-code-5537 

Author: rockstar, exarkun
Reviewer: jesstess, exarkun, dreid
Fixes: [#5537](https://github.com/twisted/twisted/issues/5537)

Delete a duplicate definition of `do_XROVER` from `NNTPServer`.  Do some code cleanups
nearby and add unit tests for this and some similar functionality, too.
